docs/CMakeLists.txt

```diff
@@ -12,9 +12,9 @@ add_custom_target(docs)
 
 option(ENABLE_DOXYGEN "Enable building doxygen documentation" ON)
```

Comment:
Shall we disable this by default with `OFF`? This should make most of the follow-up changes unnecessary.

I personally don't see the point of building doxygen locally. While it's a great feature for a website and to reference it for documentation. Many IDEs these days have a better support for documentation that don't require using doxygen in the first place.